### PR TITLE
Add partial to the enclosing types when running the ConvertToGeneratedDllImport fixer

### LINKED ISSUE #62291: DllImportGenerator conversion code fix should make the containing types partial
Right now the code fix to convert to `GeneratedDllImport` doesn't make any containing types `partial`. This can result in build failures after running the code fix (which makes the `DllImport` methods `partial`).

cc: @AaronRobinsonMSFT @elinor-fung

### PR DESCRIPTION
**Pull request by jkoritzinsky**
This also includes some refactoring to make it easier to write the custom fix all provider that was required.

Fixes #62291

### CODE REVIEW

**Comment by jkoritzinsky** [comment]
System.IO.Tests failures are in multiple PRs. Tracked in https://github.com/dotnet/runtime/issues/65791

File: src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToGeneratedDllImportAnalyzer.cs [hunk 1]
```diff
@@ -36,6 +36,9 @@ public class ConvertToGeneratedDllImportAnalyzer : DiagnosticAnalyzer
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(ConvertToGeneratedDllImport);
 
+        public const string CharSetPropertyKey = "CharSet";
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToGeneratedDllImportAnalyzer.cs:39]: Can't we `nameof` the field here?

File: src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToGeneratedDllImportFixer.cs [hunk 2]
```diff
@@ -172,38 +248,22 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
...
-                    generatedDeclaration = generator.ReplaceNode(generatedDeclaration, generatedParameterSyntax, generator.AddAttributes(generatedParameterSyntax,
-                                    GenerateMarshalAsUnmanagedTypeBoolAttribute(generator)));
-                }
-            }
-
-            if (methodSymbol.ReturnType.SpecialType == SpecialType.System_Boolean
-                && !methodSymbol.GetReturnTypeAttributes().Any(attr => attr.AttributeClass?.ToDisplayString() == TypeNames.System_Runtime_InteropServices_MarshalAsAttribute))
+        private static void MakeEnclosingTypesPartial(DocumentEditor editor, MethodDeclarationSyntax method)
+        {
+            for (TypeDeclarationSyntax typeDecl = method.FirstAncestorOrSelf<TypeDeclarationSyntax>(); typeDecl is not null; typeDecl = typeDecl.Parent.FirstAncestorOrSelf<TypeDeclarationSyntax>())
```
Comment (AaronRobinsonMSFT) [inline src/libraries/System.Runtime.InteropServices/gen/LibraryImportGenerator/Analyzers/ConvertToGeneratedDllImportFixer.cs:260]: Doesn't `typeDecl` need to be declared nullable? i.e., `TypeDeclarationSyntax?`

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/libraries/System.Runtime.InteropServices/System.Runtime.InteropServices.sln [hunk 3]
```diff
@@ -1,4 +1,8 @@
-﻿Microsoft Visual Studio Solution File, Format Version 12.00
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 17
+VisualStudioVersion = 17.2.32211.407
+MinimumVisualStudioVersion = 10.0.40219.1
```
Comment (elinor-fung) [inline src/libraries/System.Runtime.InteropServices/System.Runtime.InteropServices.sln:5]: I assume VS did this? Do we try to keep the .sln files slngen edits only?

Comment (jkoritzinsky) [reply src/libraries/System.Runtime.InteropServices/System.Runtime.InteropServices.sln:5]: Generally we try to keep the sln files such that they don't generate diffs when you open them in VS, but that usually depends on slngen generating them perfectly and sometimes it doesn't do it quite right.

File: src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/ConvertToLibraryImportFixerTests.cs [hunk 4]
```diff
@@ -481,6 +481,45 @@ public static void Code()
...
         }
 
+        [ConditionalFact]
+        public async Task MakeEnclosingTypesPartial()
+        {
+            string source = @"
+using System.Runtime.InteropServices;
+
+class Enclosing
+{
```
Comment (elinor-fung) [inline src/libraries/System.Runtime.InteropServices/tests/LibraryImportGenerator.UnitTests/ConvertToLibraryImportFixerTests.cs:491]: Maybe also have a middle or outer class that is already partial?

**Review by elinor-fung — APPROVED** [review]

**Pull request merged**